components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/bean/RuleTemplate.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.business.rules.core.bean;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Represents a Rule Template, which consists of one or more Templates
+ */
+public class RuleTemplate {
```

Comment:
Any reason against making this class (also other similar ones) immutable? 
refer: http://www.yegor256.com/2014/06/09/objects-should-be-immutable.html
e.g. https://github.com/wso2/carbon-ui-server/blob/v0.10.4/components/org.wso2.carbon.uis/src/main/java/org/wso2/carbon/uis/api/Theme.java

Reply:
This is used as a bean class. It will be instantiated through gson. Once instantiated, the state of the object won't be changed later.